### ISSUE TITLE
Request entity too large

### ISSUE THREAD

**Issue by sragneau** [node-red/node-red#1276]
We notices that if we change the url to administration route with the following parameter :+1: 
httpAdminRoot: '/admin'
we should lost the apimaxlenght value because we got  "Request entity too large".

I removed this one and no error.

**Comment by knolleary** [comment]
Hi, sorry for the delay in responding.

I'm not completely clear on what you mean here. `apiMaxLength` is used to set the maximum request size accepted by the admin api.

Are you saying that after setting `httpAdminRoot` to `/admin` the `apiMaxLength` setting did not appear to work?

**Comment by sragneau** [comment]
Yes, we did some API into the admin panel. At the beggining, i put the url of the admin panel with httpAdminRoot. I noticed that when i called an api with a big payload, i got an error entitytoolarge even if i changed the value of apiMaxLength.

So i removed httpAdminRoot paramaeer into my settings file and now apiMaxLenght is working. I can call my api with my big json body.

**Comment by iba-jsykes** [comment]
This happened to me as well, when I changed my admin root, all of a sudden posts stopped working.

When I comment it back out, no issues.  Odd.

**Comment by khelkun** [comment]
I also confirm this issue:
`httpAdminRoot: "/red"`` or anything else like `httpAdminRoot: "/admin"`` gives me the following error:
```
3 Oct 11:08:31 - [warn] [http in:Http in node] Error: request entity too large
    at readStream (/home/sebastien/gitlab-evolucare/etmi-phineus-node-red/node_modules/raw-body/index.js:196:17)
    at getRawBody (/home/sebastien/myprojectnode_modules/raw-body/index.js:106:12)
    at read (/home/sebastien/myprojectnode_modules/body-parser/lib/read.js:76:3)
    at jsonParser (/home/sebastien/myprojectnode_modules/body-parser/lib/types/json.js:127:5)
    at Layer.handle [as handle_request] (/home/sebastien/myprojectnode_modules/node-red/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/sebastien/myprojectnode_modules/node-red/node_modules/express/lib/router/route.js:137:13)
    at metricsHandler (/home/sebastien/myprojectnode_modules/node-red/nodes/core/io/21-httpin.js:215:59)
    at Layer.handle [as handle_request] (/home/sebastien/myprojectnode_modules/node-red/node_modules/express/lib/router/layer.js:95:5)
    at next (/home/sebastien/myprojectnode_modules/node-red/node_modules/express/lib/router/route.js:137:13)
    at corsHandler (/home/sebastien/myprojectnode_modules/node-red/nodes/core/io/21-httpin.js:165:48)
```
And the content-length of the HTTP request body is not that big: 367952 bytes
If I just use `httpAdminRoot: "/"`` then everything goes fine. So it's not a blocking issue.

**Comment by PotOfCoffee2Go** [comment]
I found a possible get around `Error: request entity too large` which works for me when 'httpAdminRoot' is other than '/' . 

I embed Node-RED in a `server.js` and parse the inbound JSON prior to Node-RED processing.

Note the `app.use(bodyParser.json({limit: '20mb'}));` line - which allows and parses up to 20mb of JSON content in the body. 

I assume this works because Node-RED skips parsing the body since already parsed? Anyway, when using this method - the Node-RED 'apiMaxLength' setting seems to have no effect when deploying flows - with the size limited only by the `bodyParser.json({limit: '20mb'})` setting.

```javascript
'use strict'
// {{{Modules}}}
const
    http = require('http'),
    path = require('path'),
    express = require('express'),
    bodyParser  = require('body-parser'),
    RED = require('node-red');

// {{{Node-RED settings}}}
const nodered = {
    settings: {
        httpAdminRoot:'/red',     // node-RED flow editor 
        httpNodeRoot: '/',        // node 'http in' root directory
        userDir: path.resolve(__dirname, 'node-red'), // Flow storage
        nodesDir: path.resolve(__dirname, 'node-red/nodes'), // Custom nodes
        flowFile: 'flows.json',
        flowFilePretty: true
    }
};

// {{{Host Server}}}
// Create an Express app and server
var app = express();
var server = http.createServer(app);

// Body parser to accept JSON in the body
app.use(bodyParser.json({limit: '20mb'}));

// {{{Host Routes}}}
// Serve static content(css, js, etc) from site root ('public') directory
//  without Node-RED's help :)
app.use('/', express.static('public'));

// {{{Node-RED}}}
// Initialize Node-RED runtime
RED.init(server, nodered.settings);
// Serve the Node-RED Editor and http-in node UIs
app.use(nodered.settings.httpAdminRoot, RED.httpAdmin);
app.use(nodered.settings.httpNodeRoot, RED.httpNode);
// Fire up our server
server.listen(8081);

// {{{Start Node-RED runtime}}}
RED.start();

```

**Comment by tomasliumparas** [comment]
I also came across this issue. If you set the `httpAdminRoot` lets say to `admin` and set `apiMaxLength` lets say to `10Mb` and then are trying to send a payload which is between 5 and 10  Mb you will get an error. 

However, I discovered if you change your endpoint in `http in` node to `/admin/endpoint` instead of `/ednpoint` it works as expected. I am not sure if its a bug or not?

**Comment by knolleary** [comment]
Resolved under #1278
